File: frontend/app/types/knowledge.ts
export type KnowledgeDocument = {
  readonly id: string
  readonly title: string
  readonly source_name: string
  readonly content_type: string
  readonly size_bytes: number
  readonly chunk_count: number
  readonly created_at: string
  readonly file_type: string | null
}

export type KnowledgeChunk = {
  readonly id: string
  readonly document_id: string
  readonly chunk_index: number
  readonly text: string
  readonly created_at: string
  readonly page_number: number | null
}

export type KnowledgeChunkPage = {
  readonly items: readonly KnowledgeChunk[]
  readonly total: number
  readonly offset: number
  readonly limit: number
}

export type KnowledgeSearchResult = {
  readonly id: string
  readonly document_id: string
  readonly chunk_index: number
  readonly source: string
  readonly title: string
  readonly page_number: number | null
  readonly file_type: string
  readonly text: string
  readonly distance: number | null
  readonly confidence: number
}

export type KnowledgeDeleteResult = {
  readonly status: string
  readonly message: string
}

File: backend/models/knowledge.py
from __future__ import annotations

from pydantic import BaseModel, ConfigDict, Field


class KnowledgeDocumentResponse(BaseModel):
    model_config = ConfigDict(frozen=True)

    id: str
    title: str
    source_name: str
    content_type: str
    size_bytes: int = Field(ge=0)
    chunk_count: int = Field(ge=0)
    created_at: str
    file_type: str | None = None


class KnowledgeChunkResponse(BaseModel):
    model_config = ConfigDict(frozen=True, from_attributes=True)

    id: str
    document_id: str
    chunk_index: int = Field(ge=0)
    text: str
    created_at: str
    page_number: int | None = Field(default=None, ge=1)


class KnowledgeChunkPageResponse(BaseModel):
    model_config = ConfigDict(frozen=True)

    items: list[KnowledgeChunkResponse]
    total: int = Field(ge=0)
    offset: int = Field(ge=0)
    limit: int = Field(ge=1)


class KnowledgeDeleteResponse(BaseModel):
    model_config = ConfigDict(frozen=True)

    status: str
    message: str
